### ISSUE TITLE
Chicken or the egg solution

### ISSUE THREAD

**Issue by egandro** [ArnaudRinquin/rm-local-modules#3]
HI,

I have a solution for the Chicken or the egg problem when there is no node_modules folder, yet.

    "preinstall": "rm-local-modules -v || echo \"skipting - rm-local-modules not installed yet\""

But after calling "npm install" I always have to call it again to get the local modules installed, again.

Any idea how we can fix this?

**Comment by ArnaudRinquin** [comment]
Hey @egandro I missed this issue notification and only noticed it now. Sorry about the delay.

Good spot on that issue. I'll have to fix for my own case very soon so I'll keep you posted.

**Comment by egandro** [comment]
Thank you ;) This is getting me really mad.

Also there is a related bug.


"foo": "file://../foo"

and you add a bar

"bar": "file://../bar"

it toggles ;) foo is delete, bar is created - the are not "synced".

**Comment by ArnaudRinquin** [comment]
Oh I'm curious: what version of npm do you use?

**Comment by egandro** [comment]
$ npm --version
3.10.10
$ node --version
v7.2.1

**Comment by ArnaudRinquin** [comment]
@egandro Then that's a bad case of RTFM:

> There is a caveat though, as [npm@3 preinstall is broken](https://github.com/npm/npm/issues/10379). Works fine with npm@2

npm 3 is very very broken regarding preinstall (many people are really pissed of) and there is nothing I can do.

I suggest you use `yarn`, which works just 👌

**Comment by egandro** [comment]
Sorry :(

I have now bad feelings. We are talking about 3 different isues - and - i know how to RTFM.

1) The bug I reported here - and my fix.

2) The 2nd Bug that I reported - where there is a switch of delete / reload of packages that were added after a local package is already installed.

3) The main issues on npm 3 - which I am aware of.

....

For 1 we have a fix. For 3 we can't do anything.

For 2 we have an issue rm-local-modules can take care:

If it detects "oh I killed 2 and I installed 3" it can kill all 5 and after the next "npm install" everything is fine.
